### ISSUE TITLE
No warning (CS0219) on local variables that initialized with interpolated strings, but whose values are never used

### ISSUE THREAD

**Issue by noblethrasher** [dotnet/roslyn#17154]
**Version Used**: 

Visual Studio Community 2015
Version 14.0.23107.0 D14REL

**Steps to Reproduce**:

Declare and initialize a local variable with an interpolated string, but don't subsequently use the variable. 

E.g.

    static void Main(string[] args)
    {
         var name = "Whatever";
         var foo = "Hello, world {name}";   //warning CS2019 since 'foo' is not used.
         var bar = $"Hello, word {name}";  //no warning even though 'bar' is not used.

        return;
    }

**Expected Behavior**:

Warning: **CS0219** the variable 'bar' is assigned but its value is never used.

**Actual Behavior**:

No warning.

**Comment by 0xd4d** [comment]
I think it's because there's a call to `string.Format()`. This is consistent with similar code that also don't report this warning.

```C#
string a1 = 1.ToString();
string b1 = SomeMethod();
```

IMHO, there should be a warning message but perhaps it would be considered a breaking change.

**Comment by gafter** [comment]
Dup of #2271
